builtin/logical/pki/revocation/revoke.go

```diff
@@ -21,12 +21,17 @@ const (
 )
 
 type RevokerFactory interface {
-	GetRevoker(context.Context, logical.Storage) Revoker
+	GetRevoker(context.Context, logical.Storage) (Revoker, error)
+}
+
+type RevokeCertInfo struct {
+	RevocationTime time.Time
+	Warnings       []string
 }
 
 type Revoker interface {
-	RevokeCert(cert *x509.Certificate) (*logical.Response, error)
-	RevokeCertBySerial(serial string) (*logical.Response, error)
+	RevokeCert(cert *x509.Certificate) (RevokeCertInfo, error)
+	RevokeCertBySerial(serial string) (RevokeCertInfo, error)
```

Comment:
This is a good improvement.  I didn't like the original signature but it was expedient.  Fully expected to revisit this, and in general it'd still be nice to refactor the revocation logic out of backend and into the revocation package, but that was pretty gnarly to start.